### PR TITLE
[Sync Provider] Fix panic on not found pod status

### PR DESCRIPTION
**Pull request by cpuguy83**

### CODE REVIEW

File: node/sync.go [hunk 1]
```diff
@@ -202,6 +204,11 @@ func (p *syncProviderWrapper) updatePodStatus(ctx context.Context, podFromKubern
 		log.G(ctx).Debug("Setting pod not found on pod status")
 	}
 
+	if podStatus == nil {
+		log.G(ctx).WithError(statusErr).Debug("Skipping pod status update due to missing pod status")
```
Comment (sargun) [inline node/sync.go]: This seems like a higher-level message than debug?

Comment (cpuguy83) [reply node/sync.go]: Sure, I updated this to actually return the error.

File: node/sync.go [hunk 2]
```diff
@@ -176,32 +179,35 @@ func (p *syncProviderWrapper) updatePodStatus(ctx context.Context, podFromKubern
 		return nil
 	}
 
+	if podFromKubernetes.Status.Phase != corev1.PodRunning && time.Since(podFromKubernetes.ObjectMeta.CreationTimestamp.Time) <= time.Minute {
+		span.SetStatus(err)
```
Comment (sargun) [inline node/sync.go]: Shouldn't this be `span.SetStatus(statusErr)`, because `err` gets rewritten by `cache.MetaNamespaceKeyFunc`?

**Comment by sargun** [comment]
It might be worth adding "Don't return nil pod status" to the docs.

File: node/sync.go (hunk 2 above)
Comment (cpuguy83) [reply node/sync.go]: :( Yes. Thanks

**Comment by cpuguy83** [comment]
Yeah. In this case at least it's an error that's returned with the nil status I just messed up here.

**Pull request merged**